src/fundus/publishers/de/dw.py

```diff
@@ -13,12 +13,19 @@
     generic_date_parsing,
     generic_text_extraction_with_css,
     generic_topic_parsing,
+    strip_nodes_to_text,
 )
 
 
 class DWParser(ParserProxy):
     class V2(BaseParser):
-        _paragraph_selector = CSSSelector("div.rich-text > p")
+        VALID_UNTIL = datetime.date(2024, 1, 18)
+
+        # There is a little author line at the end of the nodes gathered by the paragraph selector,
+        # which seems to be rather hard to omit. Some examples:
+        # AR/jj (dpa, rtr, ap), kle/jj (kna, dpa, rtr, afp), pg/AR/kle (dpa, afp),
+        # pg/AR/haz (afp, dpa), kle/haz (dpa, rtr, afp)
+        _paragraph_selector = XPath("//div[contains(@class, 'rich-text')] /p[not(em) or text()]")
```

Comment:
This might be an option
```suggestion
       _paragraph_selector = XPath("//div[contains(@class, 'rich-text')] /p[(not(em) or text()) and not(re:test(text(), '.*\((rtr, |dpa, |afp, |epd, |ap, ){0,3}(ap|rtr|dpa|afp|epd)\)$'))]"
                                    , namespaces={'re': 'http://exslt.org/regular-expressions'})
```

Reply:
I altered the regex a bit to make it independent of the actual acronyms.
```
^([a-z]{2,3}\/|[A-Z]{2,3}\/)*([a-z]{2,3}|[A-Z]{2,3})\s\(([a-z]{2,3}, )*([a-z]{2,3})\)$
```

Testing this, the [regex](https://regex101.com/r/uZLwyb/1) works but the selector doesn't. Can you confirm that? Did `re:test` work on your site?

Reply:
That looks good. It works for me when I add parentheses around the or statement:
`_author_regex = r"^([a-z]{2,3}\/|[A-Z]{2,3}\/)*([a-z]{2,3}|[A-Z]{2,3})\s\(([a-z]{2,3}, )*([a-z]{2,3})\)$"
_paragraph_selector = XPath(
    f"//div[contains(@class, 'rich-text')] /p[(not(em) or text()) and not(re:test(text(), '{_author_regex}'))]",
    namespaces={"re": "http://exslt.org/regular-expressions"},
)`